### DISCUSSION TITLE
List of Placeholders to use

### DISCUSSION THREAD

**Discussion by Andre601** [community/community#25095] [Actions]
<p>I would like to add a Discord notification system with GitHub Actions, to notify others about new releases I publish on the repository.</p>
<p>I use <a href="https://github.com/rjstone/discord-webhook-notify" rel="nofollow noopener">rjstone/discord-webhook-notify</a> for this.</p>
<p>The thing now is, that I want to display details from the release within the webhook message.</p>
<p>I know that there are placeholders for several things, but I’m unable to find any kind of cheatsheet to see what placeholders are available for what values.</p>
<p>Like I mentioned do I trigger this action on a Release, meaning I want to receive the title, version tag and description of said release, to use.</p>
<p>Here’s my current workflow I setup:</p>
<pre><code>name: Publication task

on:
  release:
    types: [published]
    branches:
    - master

jobs:
  publish:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v1
    - name: Set up JDK 1.8
      uses: actions/setup-java@v1
      with:
        java-version: 1.8
    - name: Publish (Bintray) # Used to publish to bintray.
      env:
        bintrayUser: ${{ secrets.bintrayUsername }}
        bintrayApiKey: ${{ secrets.bintrayApiKey }}
        BUILD_NUMBER: 0 # Releases are always major.minor.revision_0
      run: ./gradlew bintrayUpload
  sendMessage: # This is the notification task I want to use.
    if: success()
    runs-on: ubuntu-latest
    steps:
    - name: Discord notification
      uses: rjstone/discord-webhook-notify@v1.0.2
      with:
        webhookUrl: ${{ secrets.DISCORD_WEBHOOK }}
        severity: info
        username: New release
        description: ' **Release Information**'
        details: ''
</code></pre>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>You could get release title, version tag and description from the github.event context. Use <span class="math">{{ github.event.release.name }} to get the title, use </span>{{ github.event.release.body }} to get the release description. Please see my example:</p>
<p><img src="https://user-images.githubusercontent.com/24806478/181088742-342b4115-8b8b-4c40-ae30-a2249366b37b.png" alt="release event.png" data-base62-sha1="hufAgyhpSjLJhxiVObThoNs6bnT" width="643" height="393"></p>
<p>This is my release information:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/61cfe631a7f1d386c0d7432991a4f83f80f546c2.png" data-download-href="/uploads/short-url/dXhJFvAQ8EUUrxfrI8TeCe7HOxA.png?dl=1" title="release info 0.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181088743-3639528d-f39b-4e76-9703-494e482c67d6.png" alt="release info 0.png" data-base62-sha1="dXhJFvAQ8EUUrxfrI8TeCe7HOxA" width="618" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/61cfe631a7f1d386c0d7432991a4f83f80f546c2_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">release info 0.png</span><span class="informations">692×559 18.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This is the github.event context showing in debug logs:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/21ff802034826df18fa756041554c7e4f970c88e.png" data-download-href="/uploads/short-url/4QLa9EQVBZBaCVKFWkGe5Va6uHs.png?dl=1" title="release info.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181088744-a34bc26e-0bff-4603-85ab-f1e9289ab409.png" alt="release info.png" data-base62-sha1="4QLa9EQVBZBaCVKFWkGe5Va6uHs" width="690" height="432" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/21ff802034826df18fa756041554c7e4f970c88e_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">release info.png</span><span class="informations">791×496 27.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Andre601** [comment]
<p>Thanks for the info.</p>
<p>I added this but it seems that the workflow returns the error “[error]Cannot read property ‘author’ of undefined”.</p>
<p>My guess is that this is because of webhook changes to Discord?</p>
<p>I contacted the dev of the Action and see if I get any response.</p>
